File: app/src/main/java/com/softbankrobotics/qisdktutorials/ui/tutorials/peoplecharacteristics/HumanInfoAdapter.java
package com.softbankrobotics.qisdktutorials.ui.tutorials.peoplecharacteristics;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.softbankrobotics.qisdktutorials.R;
import com.softbankrobotics.qisdktutorials.model.data.HumanInfo;

import java.util.ArrayList;
import java.util.List;

/**
 * The adapter used to show humans information.
 */
class HumanInfoAdapter extends RecyclerView.Adapter<HumanInfoViewHolder> {

    private List<HumanInfo> humanInfoList = new ArrayList<>();

    @Override
    public HumanInfoViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.human_info_layout, parent, false);
        return new HumanInfoViewHolder(view);
    }

    @Override
    public void onBindViewHolder(HumanInfoViewHolder holder, int position) {
        HumanInfo humanInfo = humanInfoList.get(position);
        holder.bind(humanInfo);
    }

    @Override
    public int getItemCount() {
        return humanInfoList.size();
    }

    void updateList(List<HumanInfo> humanInfoList) {
        this.humanInfoList = humanInfoList;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/softbankrobotics/qisdktutorials/ui/tutorials/perceptions/humanawareness/HumanInfoAdapter.kt
/*
 * Copyright (C) 2018 Softbank Robotics Europe
 * See COPYING for the license
 */

package com.softbankrobotics.qisdktutorials.ui.tutorials.perceptions.humanawareness

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

import com.softbankrobotics.qisdktutorials.R

import java.util.ArrayList

/**
 * The adapter used to show humans information.
 */
internal class HumanInfoAdapter : RecyclerView.Adapter<HumanInfoViewHolder>() {

    private var humanInfoList: List<HumanInfo> = ArrayList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HumanInfoViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.human_info_layout, parent, false)
        return HumanInfoViewHolder(view)
    }

    override fun onBindViewHolder(holder: HumanInfoViewHolder, position: Int) {
        val humanInfo = humanInfoList[position]
        holder.bind(humanInfo)
    }

    override fun getItemCount() = humanInfoList.size

    fun updateList(humanInfoList: List<HumanInfo>) {
        this.humanInfoList = humanInfoList
        notifyDataSetChanged()
    }
}
